cstore_reader.c

```diff
@@ -200,6 +214,85 @@ CStoreReadFooter(StringInfo tableFooterFilename)
 }
 
 
+/* DeserializeBlockData  deserializes requested data block for all columns. It
+ * uncompresses serialized data if necessary, sets deserialized buffer index at
+ * read state. The function also deallocates previous data buffer used to keep temporary
+ * uncompressed data buffer.
+ */
+static void
+DeserializeBlockData(TupleDesc tupleDescriptor, StripeData *stripeData,
+					 MemoryContext stripeReadContext, uint64 blockIndex,
+					 BlockDataBuffers **blockDataBuffersArray)
+{
+	int columnIndex = 0;
+	MemoryContext oldContext = MemoryContextSwitchTo(stripeReadContext);
+
+	for (columnIndex=0; columnIndex < stripeData->columnCount; columnIndex++)
+	{
+		ColumnData *columnData = stripeData->columnDataArray[columnIndex];
+
+		if (columnData != NULL)
+		{
+			ColumnBlockData *blockData = columnData->blockDataArray[blockIndex];
+			Form_pg_attribute *attributeFormArray = tupleDescriptor->attrs;
+			bool typeByValue = attributeFormArray[columnIndex]->attbyval;
+			int typeLength = attributeFormArray[columnIndex]->attlen;
+			char typeAlign = attributeFormArray[columnIndex]->attalign;
```

Comment:
It seems that we don't need ```attributeFormArray``` for anything other than getting ```typeByValue```, ```typeLength```, and ```typeAlign```. So I think we can also something like this which is a bit simpler:

```C
Form_pg_attribute attributeForm = tupleDescriptor->attrs[columnIndex];
bool typeByValue = attributeForm.attbyval;
int typeLength = attributeForm.attlen;
char typeAlign = attributeForm.attalign;
```

Since ```attributeForm.attbyval```, etc. is short enough we could also have remove the three variables (```typeByValue```, etc.) we have used.